activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -124,6 +124,9 @@ def columns(table_name)
       #   column_exists?(:suppliers, :name)
       #
       #   # Check a column exists of a particular type
+      #   #
+      #   # This works for standard non-casted types (eg. string) but is unreliable
+      #   # for types that may get chasted (eg. char).
```

Comment:
what is `chasted`? Did you mean `casted`? Also what unreliable means? Why it doesn't work. I think we should explain this better. I read this, and related issue and I still don't understand what is being talked here.

Is it because the type we check against is stored as `int (6)` instead of `bigint` in the database?

Reply:
🤦  yep that's a typo. Will fix.

> Is it because the type we check against is stored as `int (6)` instead of `bigint` in the database?

Yeah that's right.

https://github.com/rails/rails/issues/43054 has a good example. Here's what happens, using Postgres or Sqlite:

```ruby
  def test_column_exists
    conn.add_column(:posts, :happy_number, :bigint)

    puts conn.column_exists?(:posts, :happy_number) # true
    puts conn.column_exists?(:posts, :happy_number, :bigint) # false
    puts conn.columns(:posts).find {|c| c.name.to_sym == :happy_number}.type # integer
    puts conn.column_exists?(:posts, :happy_number, :integer) # true - this works, but it's surprising, as the type given here doesn't match the type given to add_column
  end
```